src/main/scala/com/databricks/spark/csv/DefaultSource.scala

```diff
@@ -67,6 +67,15 @@ class DefaultSource
       throw new Exception("Escape character cannot be more than one character.")
     }
 
+    val comment = parameters.getOrElse("comment", "#")
+    val commentChar: Character = if (comment == null) {
```

Comment:
Since you used `getOrElse` on the previous line, isn't it now impossible for `comment` to be `null`?